### PR TITLE
ws: Fix test-remotectlcertificate with sscg

### PR DESCRIPTION
**Pull request by martinpitt**
If sscg is present in the build environment, the test failed on two
things:

 * The "falling back to openssl" message does not happen for
   /remotectl-certificate/create, as sscg is actually being used.
   But it will still happen for the /create-no-permission test, so make
   it optional.

 * There is a second file in  ws-certs.d/ for the sscg-generated CA.
   Expect that iff sscg is installed.

### CODE REVIEW

**Comment by martinpitt** [comment]
This was a side issue in PR #14293. For testing, install sscg in your build environment and run ./test-remotectlcertificate.

We don't test with sscg in CI right now, but we could add it to one of our images (Fedora most probably).

File: src/ws/test-remotectlcertificate.c [hunk 1]
```diff
@@ -189,6 +190,8 @@ test_valid_selfsigned (TestCase *test,
   g_assert_no_error (error);
   fname = g_dir_read_name (dir);
   g_assert_cmpstr (fname, ==, "0-self-signed.cert");
+  if (sscg_path)
+      g_assert_cmpstr (g_dir_read_name (dir), ==, "0-self-signed-ca.pem");
```
Comment (allisonkarlitskaya) [inline src/ws/test-remotectlcertificate.c:194]: "The order of entries returned from this function is not defined, and may vary by file system or other operating-system dependent factors."

Also: what about the case that both openssl and sscg are installed?  Won't we default to using openssl in that case, and then this file won't be here?

Comment (martinpitt) [reply src/ws/test-remotectlcertificate.c:194]: Good point about the entry order, I'll fix that.

If sscg is installed, it is always used and prefered. So if the -ca file is *not* there, then cert generation with sscg failed, which is exactly what I want to test.

Comment (martinpitt) [reply src/ws/test-remotectlcertificate.c]: Fixed to get along with either entry order.

**Comment by martinpitt** [comment]
Semaphore [passed](https://semaphoreci.com/cockpit-project/cockpit/branches/pull-request-14307/builds/3), but somehow didn't report back the result to GitHub.

File: src/ws/test-remotectlcertificate.c [hunk 2]
```diff
@@ -188,6 +189,15 @@ test_valid_selfsigned (TestCase *test,
   dir = g_dir_open (test->cert_dir, 0, &error);
   g_assert_no_error (error);
   fname = g_dir_read_name (dir);
+  if (sscg_path)
```
Comment (allisonkarlitskaya) [inline src/ws/test-remotectlcertificate.c:192]: this is gross, but I guess it works...

could you be convinced to try a different approach here?

 - iterate the directory, appending to a gstring, then compare both orders?
 - add items to a set-mode hash table?
 - have two 'seen' bools for each of the filenames we are expecting, set them from a loop, check them after?
 - some sorted variant of any of the above?

**Pull request merged**